### PR TITLE
Rewrite custom require hook & source map tests

### PR DESCRIPTION
**Pull request by novemberborn**
This is a rewrite of the source map tests so I can add the test coverage required in #66. This necessitated replacing the Babel test with a better one for generic custom require hooks.

This uses a new npm package named [source-map-fixtures](https://github.com/novemberborn/source-map-fixtures). It provides a few source files with different source map variants, making it easier to generate fixtures. I'm planning on updating the source map tests in AVA as well.

I've solved the problem of generating coverage reports for testing by adding a `_generateCoverage` script in `test/fixtures`. It generates a report for the source map fixtures.

And finally I've excluded the two tests from the coverage report. Seemed silly to get missing test coverage for the tests…

### CODE REVIEW

**Comment by bcoe** [comment]
\o/ this is looking great, and is very much appreciated. let me know when you need a thorough review.

**Comment by novemberborn** [comment]
@bcoe I'm finished with this now, thanks.

File: test/fixtures/_generateCoverage.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+nyc.exclude = []
+
+// Require the fixture so nyc can instrument it, then run it so there's code
+// coverage.
+fixture.require().run()
+
+// Write the coverage file so reports can be loaded.
+nyc.writeCoverageFile()
+
+var reports = _.values(nyc._loadReports()[0])
```
Comment (bcoe) [inline test/fixtures/_generateCoverage.js]: this is a great approach for generating the fixtures, and I love the module `source-map-fixtures`.

**Comment by bcoe** [comment]
The tests run way faster without the babel step, this is great.

**Pull request merged**